### PR TITLE
Logout

### LINKED ISSUE #870: Logout Page
The logout link will be available in the header of authenticated pages and when clicked the user will be redirected to a logged-out page and their session will be terminated.  The [logged-out](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=659%3A1035) page will be located at `{base_url}/auth/logout`.

To end the session the logout link will call `signOut({callbackUrl:”/logout”})`.  See Next Auth [‘signOut’](https://next-auth.js.org/getting-started/client#signout) documentation for more details.

## Design for Logged-out page:
- [Page](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=659%3A1035)
- [Header Example](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=810%3A1226)

## Acceptance Criteria:
- [x] Component follows design as presented in Figma
- [x] Component is accessible
- [x] Logged out page has path back to Login

### PR DESCRIPTION
**Pull request by falila**
# Summary | Résumé
- add logout page 
- add sign in or sign out menu
closes #870 

| After                                        |
| -----------------------------------------------
Screen Shot 2022-07-27 at 9 37 23 AM

# Test instructions | Instructions pour tester la modification
### Menu -> Sign in
- Top right corner click on sign-in 
- you should land on the [Login page](https://github.com/cds-snc/platform-forms-client/pull/884)
### Menu -> Sign out
- Top right corner click sign out 
- logout page shall be rendered. 

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

- It wasn't clear in which circumstances the `sign in or sign out` link should be displayed. By default, it will be displayed all time. Therefore, we'll end up getting two nav (login/logout) if log-in as admin. 
- The menu design should be improved @connorscarolyns and i will sync to address the styling gap. 
- We may consider a redesign of the header section. I found it really difficult to apply simple style or tailwind without fighting against `flex` which i don't  understand quite well some concepts. It might be a way to fix the banner and Fip component and make working on the header much easier. 

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/auth/LoginMenu.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import LoginMenu from "./LoginMenu";
+
+jest.mock("next-i18next", () => ({
```
Comment (jeberhardt) [inline components/auth/LoginMenu.test.js:5]: I think this should be added to the `~/__utils__/setupTests.ts` file.

File: components/auth/LoginMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const { i18n, t } = useTranslation("common");
+  const handleClick = () => {
+    signOut({ callbackUrl: "auth/logout" });
+  };
+
+  return (
+    <>
+      <div className="gc-login-menu">
+        {isAuthenticated ? (
+          <button onClick={handleClick} lang={i18n.language}>
```
Comment (jeberhardt) [inline components/auth/LoginMenu.tsx:21]: I think this requires `type="button"` for a11y requirements.

File: pages/auth/logout.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const Logout = () => {
+  const { i18n, t } = useTranslation("logout");
+  const { data: session } = useSession();
+
+  return (
+    <>
+      <div>
+        <h2 className="gc-h2">{t("messageContent")}</h2>
+        <div className="gc-last-login-time">
+          {t("lastLoginTime")} : {session?.user?.name}
```
Comment (craigzour) [inline pages/auth/logout.tsx:17]: What are we trying to display here? If I see the screenshot attached to the PR it should be a date but here it looks like you are displaying a username.

File: public/static/locales/fr/logout.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "messageContent": "Vous êtes déconnecté.",
+    "goToLoginLabel": "connectez-vous",
+    "lastLoginTime": "Votre derniere session de connexion",
```
Comment (craigzour) [inline public/static/locales/fr/logout.json:4]: `dernière` instead of `derniere` (missing accent)

File: public/static/locales/fr/logout.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "messageContent": "Vous êtes déconnecté.",
+    "goToLoginLabel": "connectez-vous",
```
Comment (craigzour) [inline public/static/locales/fr/logout.json:3]: Missing "C" at the beginning?

**Review by craigzour — CHANGES_REQUESTED** [review]

File: pages/auth/logout.tsx (hunk 3 above)
Comment (falila) [reply pages/auth/logout.tsx:17]: @craigzour  Please refer to the Unresolved  section. I've addressed this here.

**Review by jeberhardt — APPROVED** [review]

**Review by craigzour — APPROVED** [review]

**Pull request merged**